src/Raven.Server/Documents/ETL/Providers/Queue/Handlers/Processors/QueueEtlHandlerProcessorForTestAwsSqsConnection.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+using Amazon.SQS;
+using JetBrains.Annotations;
+using Newtonsoft.Json;
+using Raven.Client.Documents.Operations.ETL.Queue;
+using Raven.Server.Documents.Handlers.Processors;
+using Raven.Server.Web.System;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Server.Documents.ETL.Providers.Queue.Handlers.Processors;
+
+internal sealed class
+    QueueEtlHandlerProcessorForTestAwsSqsConnection<TRequestHandler, TOperationContext> :
+    AbstractDatabaseHandlerProcessor<TRequestHandler, TOperationContext>
+    where TOperationContext : JsonOperationContext
+    where TRequestHandler : AbstractDatabaseRequestHandler<TOperationContext>
+{
+    public QueueEtlHandlerProcessorForTestAwsSqsConnection([NotNull] TRequestHandler requestHandler) :
+        base(requestHandler)
+    {
+    }
+
+    public override async ValueTask ExecuteAsync()
+    {
+        try
+        {
+            string authenticationJson = await new StreamReader(HttpContext.Request.Body).ReadToEndAsync();
+            AwsSqsConnectionSettings connectionSettings =
+                JsonConvert.DeserializeObject<AwsSqsConnectionSettings>(authenticationJson);
+
+            IAmazonSQS client = QueueBrokerConnectionHelper.CreateAwsSqsClient(connectionSettings);
+
+            try
+            {
+                // Attempt to get the queue URL, which will validate the credentials
+                await client.GetQueueUrlAsync("connection-test");
+
+                // If we successfully get the queue URL, the credentials and permissions are valid
+                DynamicJsonValue result = new() { [nameof(NodeConnectionTestResult.Success)] = true };
+                using (ServerStore.ContextPool.AllocateOperationContext(out JsonOperationContext context))
+                await using (AsyncBlittableJsonTextWriter writer = new(context, RequestHandler.ResponseBodyStream()))
+                {
+                    context.Write(writer, result);
+                }
+            }
+            catch (AmazonSQSException ex) when (ex.ErrorCode == "QueueDoesNotExist")
+            {
+                // QueueDoesNotExist: the queue does not exist
```

Comment:
I think that `catch` code line describes this scope clearly enough 😄

Reply:
fixed